scripts/grading/grade_submission.py

```
@@ -1470,10 +1751,51 @@ def run_grading(
+            for idx, c in enumerate(raw_criteria, 1):
+                name = str(c.get("criterion_name", "")).strip()
+                pts = float(c.get("max_points", 0))
+                items = [str(i).strip() for i in c.get("checklist_items", []) if str(i).strip()]
+                if name and pts > 0 and items:
+                    rubric_criteria.append({
+                        "criterion_id": f"C{idx}",
+                        "criterion_name": name,
+                        "max_points": pts,
+                        "checklist_items": items,
+                    })
+        except Exception as exc:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)